crates/sui-benchmark/src/bin/stress.rs

```diff
@@ -536,17 +567,55 @@ async fn main() -> Result<()> {
         client_runtime.block_on(async move {
             let committee = gateway_config.make_committee().unwrap();
             let authority_clients = gateway_config.make_authority_clients();
-            let metrics = AuthAggMetrics::new(&prometheus::Registry::new());
+            let registry = metrics::start_prometheus_server(
+                format!("127.0.0.1:{}", opts.client_metric_port)
+                    .parse()
+                    .unwrap(),
+            );
+            let metrics = AuthAggMetrics::new(&registry);
             let aggregator = AuthorityAggregator::new(committee, authority_clients, metrics);
             let mut workload = make_workload(primary_gas_id, owner, keypair, &opts);
             workload.init(&aggregator).await;
             let barrier = Arc::new(Barrier::new(opts.num_workers as usize));
+            let metric = Arc::new(Metrics {
+                num_success: register_int_counter_with_registry!(
```

Comment:
For Readability's sake it might be good to introduce a constructor for this in an impl block

Reply:
done